### PR TITLE
MONGOID-4649: srv uri support

### PR DESCRIPTION
**Pull request by berga**
closes bug updating uri parser from latest mongo (> 2.5)

### CODE REVIEW

**Comment by saghm** [comment]
Thanks for the pull request! It looks like some of the changes we've made in the driver since the last release broke one of the tests on Travis, so I'm going to try to push a commit that fixes them

**Comment by saghm** [comment]
Looks like it's fixed now!

File: .ruby-version [hunk 1]
```diff
@@ -1 +0,0 @@
-2.2.1
```
[the remark this answers is not in the document]
Comment (p-mongo) [reply .ruby-version]: Yep don't need it.

File: spec/mongoid/clients/factory_spec.rb [hunk 2]
```diff
@@ -120,8 +120,8 @@
 
             let(:config) do
               {
-                default: { hosts: [ "127.0.0.1:27017" ], database: database_id },
-                secondary: { uri: "mongodb://127.0.0.1:27017,127.0.0.1:27018/mongoid_test" }
+                default: { hosts: [ "127.0.0.1:30000" ], database: database_id },
```
[the remark this answers is not in the document]
Comment (p-mongo) [reply spec/mongoid/clients/factory_spec.rb]: :+1:

**Comment by saghm** [comment]
@berga thanks for the contribution!

**Pull request merged**